File: app/src/main/java/com/ramez/shopp/Activities/ChangeLanguageActivity.java
package com.ramez.shopp.Activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.ramez.shopp.ApiHandler.DataFeacher;
import com.ramez.shopp.Classes.Constants;
import com.ramez.shopp.Classes.SettingModel;
import com.ramez.shopp.Classes.UtilityApp;
import com.ramez.shopp.Models.ResultAPIModel;
import com.ramez.shopp.databinding.ActivityChangeLanguageBinding;

public class ChangeLanguageActivity extends ActivityBase {
    private ActivityChangeLanguageBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityChangeLanguageBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        binding.chooseLangTv.setOnClickListener(view1 -> {
            binding.langContainer.setVisibility(View.VISIBLE);
            binding.chooseLangTv.setVisibility(View.GONE);

        });

        binding.containerArabic.setOnClickListener(view1 -> {
            binding.imgEnglishTick.setVisibility(View.INVISIBLE);
            binding.imgArabicTick.setVisibility(View.VISIBLE);
            UtilityApp.setLanguage(Constants.Arabic);
            UtilityApp.setAppLanguage();
            getSetting();

        });

        binding.containerEnglish.setOnClickListener(view1 -> {
            binding.imgEnglishTick.setVisibility(View.VISIBLE);
            binding.imgArabicTick.setVisibility(View.INVISIBLE);
            UtilityApp.setLanguage(Constants.English);
            UtilityApp.setAppLanguage();
            getSetting();

        });

        binding.toolBar.backBtn.setVisibility(View.GONE);

    }

    private void ChooseWelcomeActivity() {
//        Intent intent = new Intent(getActiviy(), WelcomeActivity.class);
//        startActivity(intent);
////        finish();

    }

    public void getSetting() {

        new DataFeacher(false, (obj, func, IsSuccess) -> {
            ResultAPIModel<SettingModel> result = (ResultAPIModel<SettingModel>) obj;

            if (IsSuccess) {
                SettingModel settingModel = new SettingModel();
                if (result.data != null) {
                    settingModel.setAbout(result.data.getAbout());
                    settingModel.setConditions(result.data.getConditions());
                    settingModel.setPrivacy(result.data.getPrivacy());
                    UtilityApp.setSetting(settingModel);
                    navigateChooseCityActivity();
                }

            } else {

                navigateChooseCityActivity();

            }


        }).getSetting();
    }

    public void navigateChooseCityActivity() {
        startActivity(new Intent(ChangeLanguageActivity.this, ChooseCityActivity.class));
        finish();
    }

}

File: app/src/main/java/com/ramez/shopp/activities/ChangeLanguageActivity.kt
package com.ramez.shopp.activities


import android.content.Intent
import android.os.Bundle
import android.view.View
import com.ramez.shopp.ApiHandler.DataFeacher
import com.ramez.shopp.ApiHandler.DataFetcherCallBack
import com.ramez.shopp.classes.Constants
import com.ramez.shopp.Models.SettingModel
import com.ramez.shopp.classes.UtilityApp
import com.ramez.shopp.Models.ResultAPIModel
import com.ramez.shopp.databinding.ActivityChangeLanguageBinding


class ChangeLanguageActivity : ActivityBase() {
    private lateinit var binding: ActivityChangeLanguageBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityChangeLanguageBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.chooseLangTv.setOnClickListener {
            binding.langContainer.visibility = View.VISIBLE
            binding.chooseLangTv.visibility = View.GONE
        }
        binding.containerArabic.setOnClickListener {
            binding.imgEnglishTick.visibility = View.INVISIBLE
            binding.imgArabicTick.visibility = View.VISIBLE

            navigateChooseCityActivity(Constants.Arabic)


        }
        binding.containerEnglish.setOnClickListener {
            binding.imgEnglishTick.visibility = View.VISIBLE
            binding.imgArabicTick.visibility = View.INVISIBLE

            navigateChooseCityActivity(Constants.English)

        }
        binding.toolBar.backBtn.visibility = View.GONE
    }


    val setting: Unit
        get() {
            DataFeacher(false, object :DataFetcherCallBack {
                override fun Result(obj: Any?, func: String?, IsSuccess: Boolean) {
                    val result = obj as ResultAPIModel<SettingModel?>?
                    if (IsSuccess) {
                        if (result?.data != null) {
                            val settingModel = SettingModel()
                            settingModel.about = result.data?.about + ""
                            settingModel.conditions = result.data?.conditions + ""
                            settingModel.privacy = result.data?.privacy + ""
                            UtilityApp.setSetting(settingModel)
                        }
                    }             }

            }).getSetting()
        }

    fun navigateChooseCityActivity(lang:String) {
        UtilityApp.setLanguage(lang)
        UtilityApp.setAppLanguage()
        setting
        startActivity(Intent(this@ChangeLanguageActivity, ChooseCityActivity::class.java))
        finish()
    }
}